python/lsst/ts/externalscripts/maintel/tma/random_walk_and_take_image_gencam.py

```diff
@@ -0,0 +1,583 @@
+                new_az = current_az - offset_az
+
+            # Confirm offset on sky
+            current_radec = self.tcs.radec_from_azel(az=current_az, el=current_el)
+            new_radec = self.tcs.radec_from_azel(az=new_az, el=new_el)
+            sky_offset = current_radec.separation(new_radec).value
+
+            self.log.info(
+                f"{counter=:>06d}, "
+                f"{current_az=:>.3f}, "
+                f"{new_az=:>.3f}, "
+                f"{current_el=:>.3f}, "
+                f"{new_el=:>.3f}, "
+                f"{sky_offset=:.3f}, "
+            )
+
+            # Yield the sky offset for testing purposes and
+            # and the counter to be displayed in the checkpoints
+            yield RandomWalkData(
+                counter=counter, az=new_az, el=new_el, offset=sky_offset
+            )
+
+            counter += 1
+            current_az, current_el = new_az, new_el
+
+    async def run(self):
+        tasks = [gencam.start_task for gencam in self.gencam_list]
+        await asyncio.gather(self.tcs.start_task, *tasks)
+
+        async for data in self.random_walk_azel_by_time():
+            await self.checkpoint(f"[{data.counter}] Tracking {data.az=}/{data.el=}.")
+
+            dome_encoder_position = await self.get_dome_position()
+            dome_physical_position = dome_encoder_position - self.config.dome_offset
+            self.log.debug(
+                f"Current dome encoder position: {dome_encoder_position:.3f}, "
+                f"Dome offset: {self.config.dome_offset:.3f}, "
+                f"Current dome physical position: {dome_physical_position:.3f}, "
+                f"Target telescope az: {data.az:.3f}, "
+                f"Dome Physical - Target Az: {dome_physical_position - data.az:.3f}"
+            )
+
+            if abs(dome_physical_position - data.az) > self.config.max_dome_mount_az:
+                # Move the dome and wait until it finds its final position
+                self.log.info(
+                    f"Moving from to {data.az:.3f} deg physical, "
+                    f"{data.az + self.config.dome_offset:.3f} deg encoder"
+                )
+                await self.move_dome(data.az + self.config.dome_offset)
+            else:
+                self.log.info(
+                    "The difference between Dome and TMA is too small. "
+                    "Keeping the dome at its current position."
+                )
+
+            # Slew to the next target
+            await self.slew_and_track(
+                data.az, data.el, target_name=f"random_walk_{data.counter}"
+            )
+
+            # Take data in sync with all the GenericCameras
+            await self.take_data()
+
+    def set_metadata(self, metadata):
+        """Set estimated duration of the script."""
+        metadata.duration = self.config.total_time
+
+    async def slew_and_track(self, az, el, target_name=None):
+        """Slew to and track a new target.
```

Comment:
Please expand this to explain that the tracking is in RA/Dec and az/el is just the starting point.

This does not appear try to find a star near the specified target, so I am a bit surprised that the generic camera photos reliably show anything useful. (I'm not necessary suggesting searching a catalog -- doing that well to find fields with objects of suitable brightness adds a lot of complexity, but it does make me wonder why this script takes images and if that is safe to do, given the risks of stumbling on a field with a really bright source).

Reply:
It does not always have stars in the field, and we do not care much. Maybe I will replace "target" with "field" or "position on sky." The field of these cameras are usually very large, so we usually have a couple of stars.